src/components/Username/Username.tsx

```diff
@@ -15,16 +17,27 @@ interface IUsernameProps {
  * @param length - The target length for usernames that are too long, by default 20
  */
 export const Username = ({ text, length = 20 }: IUsernameProps) => {
-  const removeKeyword = 'service-account-';
-  const resultText = text.replace(removeKeyword, '').substring(0, length);
+  const resultText = truncateUsername(text, length);
 
-  return text !== resultText ? (
+  return resultText.length !== text.length ? (
```

Comment:
This will display `..` automatically for all usernames containing `service-account-`, for example `service-account-a` will be displayed as `(SA) a..`